airframe-http-okhttp/src/main/scala/wvlet/airframe/http/okhttp/OkHttpRetryInterceptor.scala

```diff
@@ -5,19 +5,28 @@ import wvlet.airframe.control.ResultClass
 import wvlet.airframe.control.Retry.RetryContext
 import wvlet.airframe.http.HttpClientMaxRetryException
 
-import scala.util.Try
+import scala.util.control.NonFatal
+import scala.util.{Failure, Success, Try}
 
 class OkHttpRetryInterceptor(retry: RetryContext) extends Interceptor {
 
   private def dispatch(retryContext: RetryContext, chain: Interceptor.Chain): Response = {
     val request = chain.request()
 
     val response = Try(chain.proceed(request))
+    val resultClass = response match {
+      case Success(r) =>
+        try {
+          retryContext.resultClassifier(r)
+        } catch {
+          case NonFatal(e) =>
+            retryContext.errorClassifier(e)
+        }
+      case Failure(e) =>
+        retryContext.errorClassifier(e)
+    }
 
-    response.fold(
```

Comment:
Unfortunately, `fold` was not available in Scala 2.11

Reply:
I wasn't aware of that. thank you